src/averageAges.js

```diff
@@ -14,8 +14,21 @@
  *
  * @return {number}
  */
+
 function calculateMenAverageAge(people, century) {
-  // write code here
+  const men = people.filter((man) => man.sex === 'm');
+  const filteredMen = century
+    ? men.filter((man) => Math.ceil(man.died / 100) === century)
+    : men;
+
+  const ageOfMen = filteredMen.map((man) => man.died - man.born);
+
+  const totalAge = ageOfMen.reduce((prevAge, currentAge) => (
+    prevAge + currentAge
+  ), 0);
+  const averageAge = totalAge / ageOfMen.length;
+
+  return averageAge;
```

Comment:
move it in the separate function and use it here and in the calculateWomenAverageAge function